File: src/Marten.Testing/Events/fake_aggregate.js
var mt_transforms = require('mt_transforms');
mt_transforms.snapshot({
    name: 'fake_aggregate',
    $init: function () {
        return { ANames: [], BNames: [], CNames: [], DNames: [] }
    },
    event_a: function (a, e) {
        a.ANames.push(e.Name);
    },
    event_b: function (a, e) {
        a.BNames.push(e.Name);
    },
    event_c: function (a, e) {
        c.ANames.push(e.Name);
    },
    event_d: function (a, e) {
        d.ANames.push(e.Name);
    }
})

File: src/Marten.Testing/Events/FakeAggregate.cs
using System;

namespace Marten.Testing.Events
{
    public class FakeAggregate
    {
        public Guid Id { get; set; }

        public string[] ANames;
        public string[] BNames;
        public string[] CNames;
        public string[] DNames;
    }

    public class EventA
    {
        public string Name { get; set; }
    }

    public class EventB
    {
        public string Name { get; set; }
    }

    public class EventC
    {
        public string Name { get; set; }
    }

    public class EventD
    {
        public Guid Id { get; set; }

        public string Name { get; set; }
    }
}
